CRM/Contribute/BAO/Contribution.php

```diff
@@ -4161,6 +4160,27 @@ public static function getPaymentInfo($id, $component, $getTrxnInfo = FALSE, $us
           }
           $paidByLabel .= " ({$creditCardType}{$pantruncation})";
         }
+
+        // show payment edit link only for payments done via backoffice form
+        $paymentEditLink = '';
```

Comment:
Yup it was handled in https://github.com/civicrm/civicrm-core/pull/10386 which is closed and replaced by this PR